src/main/kotlin/org/rust/ide/inspections/RustNamingInspection.kt

```diff
@@ -254,7 +254,8 @@ class RustVariableNamingInspection : RustSnakeCaseNamingInspection("Variable") {
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
         object : RustElementVisitor() {
             override fun visitPatBinding(el: RustPatBindingElement) {
-                if (el.parent?.parent !is RustParameterElementImpl) {
+                val parent = el.parent?.parent
+                if (parent is RustLetDeclElement || parent is RustPatTupElement) {
```

Comment:
>r what rootPattern should do.

rootPattern should return the topmost pattern this particular binding belongs to. 

If you have something like `let (S { x }, y) = unimplemented!()` then the binding `x` is inside the pattern `S { x }` which is inside the (root) pattern `( S { x }, y)`. Not sure that "root" is the best term here though. 

The idea is that by examining the parent of the root pattern you should be able to understand the context where this bindig occurs (is it a let statement, match expression, function parameter or something else).